merlin-framework/src/main/java/gov/nasa/jpl/aerie/merlin/framework/ModelActions.java

```diff
@@ -45,27 +45,27 @@ public static void call(final String type, final Map<String, SerializedValue> ar
   }
 
   public static String defer(final Duration duration, final Runnable task) {
-    return defer(duration, threaded(task));
+    return spawn(replaying(() -> { delay(duration); spawn(task); }));
```

Comment:
If you recall, what was the motivation for `threaded(task)` in the old code? Also, what's the motivation to change to `replaying()` in the update?

Reply:
It's certainly not clear from the diff, but the `threaded` didn't go away. The inner `spawn()` invokes `spawn(Runnable)`, which itself invokes `spawn(threaded(task))`.

The outer `spawn`, which is new, replaces the primitive call to `defer` with a primitive call to `spawn()` that simply causes a delay and then another spawn. Because it's so small, we want this outer `spawn` to use a replaying task to avoid thread-switching overhead.

Reply:
The motivation for `threaded` in the old code, which remains in the new code, is that replaying semantics is surprising to modelers, who are used to methods being run from start to finish. Replaying tasks rerun earlier parts of the task more than later parts, since they need to replay up to the point they last yielded at, so logging statements and such will give surprising results. So `threaded` is simply a sane, correct default for arbitrary tasks.

Reply:
Okay, after reading your responses and looking at the code in IntelliJ with the surrounding context and thinking about it for a few minutes, I get it and am on board. 

To restate my understanding (open for correction if I have any misconceptions), we use `threaded` tasks as a default to allow callers to spawn without having to worry about the task being created, but they can also explicitly set the task type if they wish by calling `threaded()` or `replaying()`. In the case of `defer()`, that is still the case, but we're wrapping it in a replaying task with a delay first. We _could_ use a threaded task for this, but we choose `replaying` simply because we expect better performance for this limited scenario by using replaying tasks.